src/web/components/SharingPermission/bulkAddPermissionsHelpers.ts

```diff
@@ -74,6 +74,27 @@ export const hasUncheckedASharedType = (
   return false;
 };
 
+const hasPendingTypeChange = (
+  type: ClientType,
+  isChecked: boolean,
+  sharedTypes: ClientType[]
+): boolean => isChecked !== sharedTypes.includes(type);
+
+export const hasPendingTypeChanges = (
+  sharedTypes: ClientType[],
+  publisherChecked: boolean,
+  advertiserChecked: boolean,
+  DSPChecked: boolean,
+  dataProviderChecked: boolean
+): boolean => {
+  return (
+    hasPendingTypeChange('PUBLISHER', publisherChecked, sharedTypes) ||
```

Comment:
nit: I wonder if adding hasPendingTypeChange function improves any readability compared to publisherChecked !== sharedTypes.includes('PUBLISHER');

Reply:
good point, done!